src/wallet/message_transporter.cpp

```diff
@@ -192,47 +192,48 @@ bool message_transporter::delete_message(const std::string &transport_id)
   return true;
 }
 
-// Deterministically derive a transport / Bitmessage address from 'seed' (the 10-hex-digits
-// auto-config token will be used), but do not set it up for receiving in PyBitmessage as
-// well, because it's possible the address will only ever be used to SEND auto-config data
+// Deterministically derive a new transport address from 'seed' (the 10-hex-digits auto-config
+// token will be used) and set it up for sending and receiving
+// In a first attempt a normal Bitmessage address was used here, but it turned out the
+// key exchange necessary to put it into service could take a long time or even did not
+// work out at all sometimes. Also there were problems when deleting those temporary
+// addresses again after auto-config. Now a chan is used which avoids all these drawbacks
+// quite nicely.
 std::string message_transporter::derive_transport_address(const std::string &seed)
 {
+  // Don't use the seed directly as chan name; that would be too dangerous, e.g. in the
+  // case of a PyBitmessage instance used by multiple unrelated people
+  const crypto::hash &hash = crypto::cn_fast_hash(seed.data(), seed.size());
+  std::stringstream oss;
+  oss << hash;
+  std::string chan_name = oss.str().substr(1, 32);
```

Comment:
That looks like std::string chan_name = epee::string_tools::pod_to_hex(hash);

Also, does it need a salt ? luigi might have an opinion.

Reply:
Thanks, I will use the `pod_to_hex` method, overlooked that one.

About salting: Looks a bit like overkill to me, as those chan names do not appear in public anywhere, only the PyBitmessage instances of two signers get to see them.

Plus two computers would have to come up with an identical salt value independently, i.e. without any communication between them - right now I lack a good idea how to do this ...

Reply:
I meant a static salt particular to this code so that hashing the same thing in different contexts don't give the same result, like, say:

```
    static key ecdhHash(const key &k)
    {
        char data[38];
        rct::key hash;
        memcpy(data, "amount", 6);
        memcpy(data + 6, &k, sizeof(k));
        cn_fast_hash(hash, data, sizeof(data));
        return hash;
    }
```

It might indeed be overkill. I'll ping luigi to ask.

Reply:
From IRC:
<@luigi1111w> moneromooo it's probably fine, but it's very low cost to implement, so I'd be in favor of including it anyway

Reply:
I see. Done, because indeed easy.